mode/dockerfile/dockerfile.js

```diff
@@ -12,9 +12,10 @@
   "use strict";
 
   // Collect all Dockerfile directives
-  var instructions = ["from", "maintainer", "run", "cmd", "expose", "env",
+  var instructions = ["arg", "from", "maintainer", "run", "cmd", "label", "expose", "env",
                       "add", "copy", "entrypoint", "volume", "user",
-                      "workdir", "onbuild"],
+                      "workdir", "onbuild", "stopsignal", "healthcheck", "shell"],
+      instructionFromAs = /^\s*\b(?:(FROM))\b(.*?)\b(?:(AS))\b/i,
```

Comment:
I think this would not correctly handle the following:

> FROM debian:stretch # I tend to use stable as that is more stable

Reply:
I fixed it and added a testcase.